### DISCUSSION TITLE
Pattern does not endup inside schema when using regex

### DISCUSSION THREAD

**Discussion by ssbarnea** [pydantic/pydantic#2706] [Question] [answered]
I am trying to add a constrained string via https://github.com/ansible-community/schemas/pull/36/files but the generated schema (part of the same change) does not include the expected `pattern` field.

I suspect this has something to do with the fact that I am using the Union of types as this constrained string can be at the root level. Still, I was not able to find any documentation example on how to achieve this when using __root__ and apparently my attempt of subclassing the ConstrinedString did not had the desired outcome.

```
class EncryptedString(str):
    regex: Optional[Pattern[str]] = constr(regex=r'^apple (pie|tart|sandwich)$')


class VarsModel(BaseModel):
    __root__: Union[Mapping[str, Any], EncryptedString]
```

Any hints on how to do this correctly? Thanks.

**Comment by PrettyWood** [comment, marked as answer]
Hi @ssbarnea 
Your first class mixes `str` and `constr`. You can either create a `BaseModel` (but overkilled) or directly create a custom type
```py
EncryptedString = constr(regex=r'^apple (pie|tart|sandwich)$')
```
or
```py
class EncryptedString(ConstrainedStr):
    regex = re.compile(r'^apple (pie|tart|sandwich)$')
```
to get
```
{
  "title": "VarsModel",
  "anyOf": [
    {
      "type": "object"
    },
    {
      "type": "string",
      "pattern": "^apple (pie|tart|sandwich)$"
    }
  ]
}
```
Is that what you want?

**Reply by ssbarnea** [reply]
Thanks! That was my intent.

As a note, the first example does make mypy unhappy, generating an error:
```
error: Variable "ansibleschemas.vars.EncryptedString" is not valid as a type
note: See https://mypy.readthedocs.io/en/latest/common_issues.html#variables-vs-type-aliases
```

Still, the second one works just fine.
